### PR TITLE
Messenger: fix for interface_exists

### PR DESCRIPTION
**Pull request by weaverryan**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no->
| Tests pass?   | yes
| Fixed tickets | none
| License       | MIT
| Doc PR        | not needed

2 unrelated bugs! But, very minor :).

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -347,8 +347,6 @@ public function load(array $configs, ContainerBuilder $container)
             ->addTag('validator.initializer');
         $container->registerForAutoconfiguration(ReceiverInterface::class)
             ->addTag('messenger.receiver');
-        $container->registerForAutoconfiguration(SenderInterface::class)
```
Comment (sroze) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Why would you remove that? 🤨

**Review by sroze — COMMENTED** [review]
Thank you very much for the `interface_exists`, l was meant to look at why it was auto enabling it today hehe. But... the Sender is valid isn’t it?

**Review by sroze — APPROVED** [review]
Thanks!

**Comment by sroze** [comment]
@weaverryan I've updated the tests within your PR as well.

**Review by fabpot — APPROVED** [review]

**Comment by sroze** [comment]
Though, it breaks the tests with a service that has a dependency on the `logger` service. The service is removed via the `MessengerPass`. Maybe the best idea is to make the `logger` dependency lazy with `on-invalid="null"` on this given service?

**Review by xabbuh — APPROVED** [review]

**Comment by sroze** [comment]
Thank you @weaverryan.

**Pull request merged**